### PR TITLE
Add Parameter.assign docstring

### PR DESCRIPTION
**Pull request by awav**
Addressing #1191

### CODE REVIEW

**Comment by st--** [comment]
Thanks for the docstring improvement! To actually close #1191 I think we still need to change cell [10] in [doc/source/notebooks/understanding/models.ipynb](https://github.com/GPflow/GPflow/blob/develop/doc/source/notebooks/understanding/models.ipynb) from `m.kernel.kernels[0].lengthscale.assign(new_parameter)` to `m.kernel.kernels[0].lengthscale = new_parameter`. Do you want to do that in this PR ?

**Comment by awav** [comment]
@st-- , yep, will do that!

File: gpflow/base.py [hunk 1]
```diff
@@ -112,8 +112,27 @@ def trainable(self, flag: Union[bool, int]):
...
+            ```
+
+        :param value: Constrained tensor-like value.
+        :param use_locking: If `True`, use locking during the assignment.
+        :param name: The name of the operation to be created.
+        :param read_value: if True, will return something which evaluates to the new
+            value of the variable; if False will return the assign op.
+        """
         value = _verified_value(value, self.dtype)
         unconstrained_value = _to_unconstrained(value, self.transform)
```
Comment (st--) [inline gpflow/base.py]: Is this missing a `return` on the call to `self._unconstrained.assign()` below?

File: doc/source/notebooks/understanding/models.ipynb [hunk 2]
```diff
@@ -24,7 +24,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": 27,
```
Comment (st--) [inline doc/source/notebooks/understanding/models.ipynb]: Would be great if you could "Restart&Run all" the notebook to make sure it all runs correctly & so the cells are numbered in a nice sequence!

Comment (awav) [reply doc/source/notebooks/understanding/models.ipynb]: done

File: gpflow/base.py (hunk 1 above)
Comment (awav) [reply gpflow/base.py]: Yes, it is. It was intentional, but for protocol compatibility, I added it back.

**Comment by awav** [comment]
@st--, Thanks! I addressed your comments! Ready to merge it, waiting for tests to pass.

**Pull request merged**